### PR TITLE
[RFR] Improve form

### PR DESCRIPTION
**Pull request by BaptistePDJ**

### CODE REVIEW

File: project/src/Controller/ClientController.php [hunk 1]
```diff
@@ -5,6 +5,7 @@
 use App\Entity\Client;
 use App\Form\ClientType;
 use App\Repository\ClientRepository;
+use App\Repository\SiteRepository;
 use Doctrine\ORM\EntityManagerInterface;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
 use Symfony\Component\HttpFoundation\{Request, Response};
```
Comment (BaptistePDJ) [inline project/src/Controller/ClientController.php]: you can merge some

File: project/src/Controller/ServerController.php [hunk 2]
```diff
@@ -5,6 +5,7 @@
 use App\Entity\Server;
 use App\Form\ServerType;
 use App\Repository\ServerRepository;
+use App\Repository\SiteRepository;
 use Doctrine\ORM\EntityManagerInterface;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
 use Symfony\Component\HttpFoundation\{Request, Response};
```
Comment (BaptistePDJ) [inline project/src/Controller/ServerController.php]: merge

File: project/src/Controller/ServerController.php [hunk 3]
```diff
@@ -80,11 +80,15 @@ public function edit(Request $request, Server $server, EntityManagerInterface $e
...
+            $server = $form->getData();
+            foreach ($form->get('site')->getData() as $site) {
+                $server->addSite($site);
+            }
+            $entityManager->persist($server);
             $entityManager->flush();
-
             return $this->redirectToRoute('server_index', [], Response::HTTP_SEE_OTHER);
         }
-
```
Comment (BaptistePDJ) [inline project/src/Controller/ServerController.php:87]: break line

File: project/src/Controller/ServerController.php [hunk 4]
```diff
@@ -80,11 +80,15 @@ public function edit(Request $request, Server $server, EntityManagerInterface $e
...
 
         if ($form->isSubmitted() && $form->isValid()) {
+            /** @var Server $server */
+            $server = $form->getData();
+            foreach ($form->get('site')->getData() as $site) {
+                $server->addSite($site);
+            }
+            $entityManager->persist($server);
             $entityManager->flush();
-
```
Comment (BaptistePDJ) [inline project/src/Controller/ServerController.php:84]: break line

File: project/src/Controller/SiteController.php [hunk 5]
```diff
@@ -2,8 +2,11 @@
...
 namespace App\Controller;
 
+use App\Entity\Server;
 use App\Entity\Site;
 use App\Form\SiteType;
+use App\Repository\DomainNameRepository;
+use App\Repository\ServerRepository;
 use App\Repository\SiteRepository;
 use Doctrine\ORM\EntityManagerInterface;
 use Symfony\Bundle\FrameworkBundle\Controller\AbstractController;
```
Comment (BaptistePDJ) [inline project/src/Controller/SiteController.php]: merge

File: project/templates/domain_name/show.html.twig [hunk 6]
```diff
@@ -46,7 +46,12 @@
                     </tr>
                 <tr>
                     <th>Site</th>
-                    <td><a href="{{ path('site_show', {'id': domain_name.site.id}) }}" class="a-link btn icon-title-left" data-toggle="tooltip" title="Cliquez pour plus de détails">{{ domain_name.site.name ? "- " ~ domain_name.site.name }}</a></td>
+                    {% if domain_name.site is defined and domain_name.site is not null %}
+                        <td>
+                            {{ domain_name.site.name ? "- " ~ domain_name.site.name }}
```
Comment (BaptistePDJ) [inline project/templates/domain_name/show.html.twig]: why a verif on .name ?

File: project/src/Controller/ClientController.php [hunk 7]
```diff
@@ -95,4 +95,16 @@ public function delete(Request $request, Client $client, EntityManagerInterface
...
+    public function removeSite(Request $request, Client $client, EntityManagerInterface $entityManager, SiteRepository $siteRepository): Response
+    {
+        $site = $siteRepository->findOneBy(['id' => $request->get('siteId')]);
+        $client->removeSite($site);
+        $entityManager->flush();
+
+        return $this->render('client/show.html.twig', [
+            'client' => $client,
+        ]);
+    }
```
Comment (julienmilazzo) [inline project/src/Controller/ClientController.php]: Why you don't do it like this : 
```
    #[Route('{clientId}/remove-site/{siteId}', name: 'client_remove_site', methods: ['GET', 'POST'])]
    public function removeSite(Request $request, Client $client, Site $site, EntityManagerInterface $entityManager): Response
    {
        $client->removeSite($site);
        $entityManager->flush();

        return $this->render('client/show.html.twig', [
            'client' => $client,
        ]);
    }
```

File: project/src/Controller/ClientController.php [hunk 8]
```diff
@@ -3,6 +3,7 @@
 namespace App\Controller;
 
 use App\Entity\Client;
+use App\Entity\Site;
```
Comment (julienmilazzo) [inline project/src/Controller/ClientController.php]: you can group this

File: project/src/Controller/ServerController.php [hunk 9]
```diff
@@ -3,6 +3,7 @@
 namespace App\Controller;
 
 use App\Entity\Server;
+use App\Entity\Site;
```
Comment (julienmilazzo) [inline project/src/Controller/ServerController.php]: same here

File: project/src/Controller/ClientController.php [hunk 10]
```diff
@@ -97,9 +98,8 @@ public function delete(Request $request, Client $client, EntityManagerInterface
     }
 
     #[Route('/remove-site/{id}', name: 'client_remove_site', methods: ['GET', 'POST'])]
```
Comment (julienmilazzo) [inline project/src/Controller/ClientController.php:102]: `#[Route('{clientId}/remove-site/{siteId}', name: 'client_remove_site', methods: ['GET', 'POST'])]`

File: project/src/Controller/SiteController.php [hunk 11]
```diff
@@ -2,6 +2,8 @@
 
 namespace App\Controller;
 
+use App\Entity\DomainName;
+use App\Entity\Server;
```
Comment (julienmilazzo) [inline project/src/Controller/SiteController.php]: same here

File: project/src/Controller/SiteController.php [hunk 12]
```diff
@@ -121,14 +122,14 @@ public function removeServer(Request $request, Site $site, EntityManagerInterfac
...
     {
-        $domainName = $domainNameRepository->findOneBy(['id' => $request->get('domainNameId')]);
         $site->removeDomainName($domainName);
         $entityManager->flush();
 
         return $this->render('site/show.html.twig', [
             'site' => $site,
         ]);
     }
+
```
Comment (julienmilazzo) [inline project/src/Controller/SiteController.php]: remove this break line pls

**Review by julienmilazzo — APPROVED** [review]

**Pull request merged**